jobfunnel/tests/test_tools.py

```diff
@@ -42,6 +44,148 @@
     }
 ]
 
+# Define a attribute list for all tests to use in this module
+
+attr_list = [
+    [['title'], 'Test Engineer'],
+    [['title'], 'Software Engineer–'],
+    [['blurb'], 'Test and develop'],
+    [['blurb'], 'Develop and design software–'],
+    [['date'], 'Just posted'],
+    [['date'], 'today'],
+    [['date'], '1 hour ago'],
+    [['date'], '2 hours ago'],
+    [['date'], 'yesterday'],
+    [['date'], '1 day ago'],
+    [['date'], '2 days ago'],
+    [['date'], '1 month'],
+    [['date'], '2 months'],
+    [['date'], '1 year ago'],
+    [['date'], '2 years ago'],
+    [['date'], '1 epoch ago'],
+    [['date'], 'junk'],
+    [['some_option'], 'option_value']
+]
+
+# Test clean/dirty characters that may be on title and blurb fields
+
+
+def test_filter_non_printables_clean_title(job_listings):
+    job_list = job_listings(attr_list[0:1])
```

Comment:
Thanks so much for pointing this out! I know it looks awkward, but the reason for that is because job_listings depends on the functionality of `config_factory` which expects a list of attributes. If I did attr_list[0], it would just be an item and not a list.

Reply:
Oh yes... I see the problem is in `config_factory`. We can leave this issue for another time. The use of this nested lists also seems a bit odd to me know but maybe this is because I haven't had the time to have a look at the code base for some time.

I live in Paris and the situation is quite dire at the moment to be honest. Most ICU capacity is in-use and other care has been postponed with possible dramatic consequences for a long time. I'm happy to hear that you and your family are being fine so far. Living in a rural area definitely helps. I do hope, however, that your community will adhere the restrictions put in place because I have first-hand experience how bad it can be if the virus spreads uncontrollably.

Reply:
Our hearts and thoughts are with everyone in Paris. I hope things get better for everyone over there soon.

Reply:
stay safe @markkvdb and @thebigG ! 

I hope that things may return to somewhat normal by summer. 

Things are still fairly quiet here in Canada - everyone is self-isolating. 

all the best. 
- Paul